### DISCUSSION TITLE
Flashing the firmware files

### DISCUSSION THREAD

**Discussion by ksjh** [ksjh/silabs-firmware-builder#5] [General]
I mainly use the [NabuCasa Universal Silicon Labs Flasher](https://github.com/NabuCasa/universal-silabs-flasher) when trying my firmware builds. There are three things to make sure:
- The ``--baudrate`` needs to be adapted to the baud rate of the firmware that is **currently** installed on the dongle.
- The options ``--allow-cross-flashing`` and ``--allow-reflash-same-version`` might be needed, depending on the currently installed firmware.
- When the new experimental OpenThread RCP builds  (Thread only, no Zigbee support) have already been installed on the dongle, the current version of the NabuCasa flasher **will not work**. So, **do not flash these versions**, unless you know what you are doing and know how to flash the dongle by other means.

**Comment by darkxst** [comment]
> * The `--baudrate` needs to be adapted to the baud rate of the firmware that is **currently** installed on the dongle.

This option has been deprecated in latest versions of Universal-silabs-flasher. Baudrate should be automatically probed now, however you can still specify app specific baudrate if really required (i.e. `--ezsp-baudrate`)

**Comment by ElVit** [comment]
Hi, is it possible to use the [Home Assistant Add-on: Silicon Labs Flasher Add-on](https://github.com/home-assistant/addons/tree/master/silabs_flasher) to flash the ZBDongle-E?
Is it the same as you meantioned above ([NabuCasa Universal Silicon Labs Flasher](https://github.com/NabuCasa/universal-silabs-flasher))?

**Reply by darkxst** [reply]
I've not tested that, however It will probably work, if you specify as suitable custom firmware. It uses the NabuCasa Universal Silabs flasher under the hood.

Note that the current ZBDongle-E builds on this repo have hardware flow control enabled, however the ZBDongle-E doesnt actually have hardware flow control.
